src/app/g3w-ol/controls/querybboxcontrol.js

```diff
@@ -1,100 +1,249 @@
 import { SPATIALMETHODS, VM } from 'g3w-ol/constants';
+import GUI from 'services/gui';
+import DataRouterService from 'services/data';
+import ProjectsRegistry from 'store/projects';
 
-const { merge } = require('core/utils/ol');
+const {throttle} = require('core/utils/utils');
 const InteractionControl = require('g3w-ol/controls/interactioncontrol');
 
-const QueryBBoxControl = function(options = {}){
-  const {spatialMethod=SPATIALMETHODS[0]} = options;
+// get all filtrable layers in toc no based on selection or visibility
+const layersFilterObject = {
+  SELECTEDORALL: true, // selected or all
```

Comment:
- if possible, I think it could be better write this in a snake case (eg: `SELECTED_OR_ALL: true,`)
- for `true / false` values it is not so easy to understand what we are referring to (e.g. true --> `selected` and false --> `all` ?)
- alternatively, it could be feasible to use two different object properties or literal values (`'selected'` | `'all'`) instead of boolean values?